controllers/kubemarkmachine_controller.go

```diff
@@ -114,24 +115,34 @@ func (r *KubemarkMachineReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		}
 	}()
 
+	kubemarkClusterClient, kubemarkClusterNamespace, err := r.KubemarkCluster.GenerateKubemarkClusterClient(kubemarkMachine.Spec.KubemarkClusterSecretRef, kubemarkMachine.Namespace, ctx)
```

Comment:
this retrieves the client:
- if new secret reference for external cluster kubeconfig/namespace is present it will validate and use that
- if new secret reference is not present it will fallback to the existing client